roles/compliance/tasks/install.yml

```diff
@@ -0,0 +1,142 @@
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Install the compliance operator
+- name: Create the compliance namespace
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    api_version: v1
+    kind: Namespace
+    state: present
+    definition:
+      kind: Namespace
+      metadata:
+        name: "{{ cifmw_compliance_namespace }}"
+        labels:
+          openshift.io/cluster-monitoring: "true"
+          pod-security.kubernetes.io/enforce: privileged
+  when: not cifmw_compliance_dry_run | bool
+
+- name: Ensure OperatorGroup for the compliance-operator is present
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit) }}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    state: present
+    wait: true
+    definition:
+      apiVersion: operators.coreos.com/v1
+      kind: OperatorGroup
+      metadata:
+        name: compliance-operator
+        namespace: "{{ cifmw_compliance_namespace }}"
+      spec:
+        targetNamespaces:
+          - "{{ cifmw_compliance_namespace }}"
+  when: not cifmw_compliance_dry_run | bool
+
+- name: Ensure Subscription for the compliance-operator is present
+  kubernetes.core.k8s:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit)}}"
+    state: present
+    wait: true
+    definition:
+      apiVersion: operators.coreos.com/v1alpha1
+      kind: Subscription
+      metadata:
+        name: compliance-operator-sub
+        namespace: "{{ cifmw_compliance_namespace }}"
+      spec:
+        name: compliance-operator
+        source: redhat-operators
+        sourceNamespace: openshift-marketplace
+        channel: "stable"
+        installPlanApproval: Automatic
+  when: not cifmw_compliance_dry_run | bool
+
+- name: Wait until the compliance-operator csv is present
+  kubernetes.core.k8s_info:
+    kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+    api_key: "{{ cifmw_openshift_token | default(omit)}}"
+    context: "{{ cifmw_openshift_context | default(omit) }}"
+    namespace: "{{ cifmw_compliance_namespace }}"
+    kind: ClusterServiceVersion
+    api_version: operators.coreos.com/v1alpha1
+  register: csv_list
+  delay: 10
+  retries: 20
+  until: >-
+    {{
+      csv_list.resources |
+      map(attribute='metadata.name') |
+      select('match', 'compliance-operator')
+    }}
```

Comment:
Can you rely on `wait` definining a `wait_condition` instead of using `until`?

Reply:
Not sure - but this is consistent with our other existing roles.  Can we defer this to a follow-up patch?

Reply:
Sure thing